components/Switch.vue

```diff
@@ -0,0 +1,102 @@
+<script>
+import { Switch as Toggle, SwitchGroup, SwitchLabel } from "@headlessui/vue";
+
+export default {
+  name: "AxoSwitch",
+  components: {
+    Toggle,
+    SwitchGroup,
+    SwitchLabel,
+  },
+  props: {
+    as: {
+      type: String,
+      required: false,
+      default: "div",
+    },
+    inversionColor: {
+      type: [String, null],
+      required: false,
+      default: null,
+    },
+    labels: {
+      type: Array,
+      required: false,
+      default: () => ["", ""],
+    },
+    toggled: {
+      type: Boolean,
+      required: false,
+      default: false,
+    },
+  },
+  emits: ["update:toggled"],
+  data() {
+    return {
+      enabled: this.toggled,
+    };
+  },
+  computed: {
+    displayText() {
+      if (Array.isArray(this.labels)) {
+        return this.enabled ? this.labels[1] : this.labels[0];
+      }
+
+      return this.labels;
+    },
+    enabledLabelClass() {
+      return [
+        { [this.inversionColor]: this.isEnabledAndInverted },
+        {
+          "text-slate-100 pt-2 pb-1 rounded-md": this.isEnabledAndInverted,
+        },
+        "font-medium px-2",
+      ];
+    },
+    enabledSwitchClass() {
+      return this.enabled ? "bg-violet-600" : "bg-slate-200";
+    },
+    enabledSpanClass() {
+      return this.enabled ? "translate-x-5" : "translate-x-0";
+    },
+    isEnabledAndInverted() {
+      return Boolean(this.inversionColor) && this.enabled;
+    },
+  },
+  watch: {
+    toggled(newEnabled) {
+      this.enabled = newEnabled;
+    },
+  },
```

Comment:
This is the only difference in this file, we need it because if there is an error I want to update the toggled value and without this the Switch will never update it

Reply:
much better! Approved

Reply:
Merged and this file was deleted :)